cloudsync/youtube.py

```diff
@@ -0,0 +1,247 @@
+""" YouTube API interface"""
+import time
+import http
+import logging
+from tempfile import NamedTemporaryFile
+
+import boto3
+import httplib2
+import oauth2client
+
+from django.conf import settings
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from googleapiclient.http import MediaFileUpload
+
+from ui.encodings import EncodingNames
+
+log = logging.getLogger(__name__)
+
+
+def resumable_upload(request):
+    """
+    Upload a video to YouTube and resume on failure up to 10 times, adapted from YouTube API example
```

Comment:
Can you add the type and return value to the docstring?

Reply:
Sure